test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java

```diff
@@ -417,6 +418,44 @@ public void testAclSqlResultSetHasOneRecord() throws SQLException {
     assertEquals(golden, acl);
   }
 
+  @Test
+  public void testSingleDocSqlResultSetHasOneRecord() throws SQLException {
+    executeUpdate("create table data("
+        + "id int, col1 varchar(20),"
+        + GsaSpecialColumns.GSA_PERMIT_GROUPS + " varchar(20),"
+        + GsaSpecialColumns.GSA_PERMIT_USERS + " varchar(20),"
+        + GsaSpecialColumns.GSA_DENY_GROUPS + " varchar(20),"
+        + GsaSpecialColumns.GSA_DENY_USERS + " varchar(20))");
+    executeUpdate("insert into data("
+        + "id, col1,"
+        + GsaSpecialColumns.GSA_PERMIT_GROUPS + ","
+        + GsaSpecialColumns.GSA_PERMIT_USERS + ","
+        + GsaSpecialColumns.GSA_DENY_GROUPS + ","
+        + GsaSpecialColumns.GSA_DENY_USERS + ") values ("
+        + "1, 'test', "
+        + "'pgroup1, pgroup2', 'puser1, puser2', "
+        + "'dgroup1, dgroup2', 'duser1, duser2')");
+    Acl golden = new Acl.Builder()
+        .setPermitUsers(Arrays.asList(
+            new UserPrincipal("puser2"),
+            new UserPrincipal("puser1")))
+        .setDenyUsers(Arrays.asList(
+            new UserPrincipal("duser1"),
+            new UserPrincipal("duser2")))
+        .setPermitGroups(Arrays.asList(
+            new GroupPrincipal("pgroup1"),
+            new GroupPrincipal("pgroup2")))
+        .setDenyGroups(Arrays.asList(
+            new GroupPrincipal("dgroup2"),
+            new GroupPrincipal("dgroup1")))
+        .build();
+    ResultSet rs = executeQuery("select * from data");
+    rs.next();
+    Acl acl = DatabaseAdaptor.buildAcl(rs, ",", DEFAULT_NAMESPACE,
```

Comment:
There are only six tests that use buildAcl instead of getDocContent. I think we should convert all of them, perhaps in a separate CL. Perhaps minimally we could use getDocContent for this new test. We have more logic around db.aclSql now, and I don't like bypassing that in the tests.

Reply:
Agree. Modified this test to call getDocContent instead of buildAcl.